src/output/file.rs

```diff
@@ -0,0 +1,289 @@
+// (c) Meta Platforms, Inc. and affiliates.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+use std::collections::BTreeMap;
+
+use crate::output as tv;
+use crate::spec;
+use maplit::{btreemap, convert_args};
+
+/// This structure represents a File message.
+/// ref: https://github.com/opencomputeproject/ocp-diag-core/tree/main/json_spec#file
+///
+/// # Examples
+///
+/// ## Create a File object with the `new` method
+///
+/// ```
+/// # use ocptv::output::*;
+///
+/// let file = File::new("name", "/path/to/file", false);
+/// ```
+///
+/// ## Create a File object with the `builder` method
+///
+/// ```
+/// # use ocptv::output::*;
+///
+/// let file = File::builder("name", "/path/to/file", false)
+///     .description("description")
+///     .content_type("text/plain")
+///     .add_metadata("key", "value".into())
+///     .build();
+/// ```
+#[derive(Default)]
+pub struct File {
+    name: String,
+    uri: String,
+    is_snapshot: bool,
+    description: Option<String>,
+    content_type: Option<String>,
+    metadata: Option<BTreeMap<String, tv::Value>>,
+}
+
+impl File {
+    /// Builds a new File object.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let file = File::new("name", "/path/to/file", false);
+    /// ```
+    pub fn new(name: &str, uri: &str, is_snapshot: bool) -> Self {
```

Comment:
`is_snapshot` is indeed mandatory in the spec, but i think we can make a decision for the user here and default it to True; The `FileBuilder` can be used to set it to an arbitrary value

additionally, it just reads badly. `new("name", "path", ... some weird bool)`

Reply:
Makes sense, but I think it should be false by default